assets/js/admin/wpjm-notice-dismiss.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import domReady from '@wordpress/dom-ready';
+
+domReady( () => {
+
+	/**
+	 * Handle dismissing the notice by sending a request to the server.
+	 *
+	 * @param  element The DOM element of the container of the notice being dismissed.
+	 */
+	const handleDismiss = ( element ) => {
+		const formData = new FormData();
+		if ( element.dataset.dismissNotice ) {
+			formData.append( 'notice', element.dataset.dismissNotice );
+		}
+		formData.append( 'action', element.dataset.dismissAction );
+		formData.append( 'nonce', element.dataset.dismissNonce );
+
+		fetch( ajaxurl, {
+			method: 'POST',
+			body: formData,
+		} );
+	};
+
+	document.body.addEventListener( 'click', ( event ) => {
```

Comment:
I imagine you used this approach because the button is being added later for some reason, right?
If yes, I wonder if we could reduce the `body` scope to something more specific. So we prevent running this function on every click on the page. WDYT?

Reply:
I used the approach in Sensei (copied from Sensei), so if we do updates here, we'll want to take a look at Sensei as well. To tighten this up, are we thinking something like this?

```
const wpjmNotice = document.querySelector('.wpjm-admin-notice' );
if ( wpjmNotice ) {
    wpjmNotice.addEventListener( 'click', ( event ) => {
    ...
}
```

Reply:
Committed https://github.com/Automattic/WP-Job-Manager/pull/2401/commits/eb913659a1f603c37f759cab5c791f44ea082b3a

Reply:
Hi @anaemnesis! 👋

Yep! Thank you for the change. But in this case, maybe we don't need the conditional anymore since we're only getting clicks under `.wpjm-admin-notice`, right? Not something that would cause a bug though. 🙂